File: java/TwoPointers/RemoveElement.java
package java.TwoPointers;

public class RemoveElement {

    /*
     * 27. Remove Element
     * Difficulty: Easy
     * Topics: Array, Two Pointers
     */
    public static int removeElement(int[] nums, int val) {
        int index = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[index] = nums[i];
                index++;
            }
        }
        return index;  
    }
}

File: python/TwoPointers/RemoveElement.py
"""
27. Remove Element
Difficulty: Easy
Topics: Array, Two Pointers
"""
def removeElement(nums: list[int], val: int) -> int:
    left, right = 0, len(nums)

    for i in range(right):
        if nums[i] != val:
            nums[left] = nums[i]
            left += 1
        
    return left